### PR TITLE
Migrate off of msRestAzure api to equivalents

### PR DESCRIPTION
**Pull request by Sandido**
---

MSRestAzure is being deprecated due to security issues. I am migrating vm repair off of that API onto different ones that do the same. 
Existing tests and manual tests pass with this change. 

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by zhoxing-ms** [comment]
@Sandido Please resole these conflicts

**Comment by Sandido** [comment]
@zhoxing-ms , please review this.

File: src/vm-repair/setup.py [hunk 1]
```diff
@@ -8,10 +8,8 @@
 from codecs import open
 from setuptools import setup, find_packages
 
-
 VERSION = "1.1.0"
```
Comment (zhoxing-ms) [inline src/vm-repair/setup.py:11]: Please upgrade the extension version in the `setup.py` to trigger the release of new extension version

**Review by zhoxing-ms — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
@Sandido Please resolve these conflicts

**Review by evelyn-ys — APPROVED** [review]

File: src/vm-repair/setup.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from codecs import open
 from setuptools import setup, find_packages
 
-VERSION = "1.1.0"
+VERSION = "1.2.0"
```
Comment (evelyn-ys) [inline src/vm-repair/setup.py:11]: As a minor change, it's better to upgrade the last version `1.1.1` instead of `1.2.0`

Comment (Sandido) [reply src/vm-repair/setup.py]: @evelyn-ys , done.

**Pull request merged**